cmd/wasm-client/main.go

```diff
@@ -281,6 +313,88 @@ func (w *wasmClient) InvokeRPC(_ js.Value, args []js.Value) interface{} {
 
 }
 
+func (w *wasmClient) GetExpiry(_ js.Value, _ []js.Value) interface{} {
+	if w.mac == nil {
+		return nil
+	}
+
+	expiry, found := checkers.ExpiryTime(nil, w.mac.Caveats())
+	if !found {
+		return nil
+	}
+
+	return js.ValueOf(expiry.Unix())
+}
+
+func (w *wasmClient) HasPermissions(_ js.Value, args []js.Value) interface{} {
+	if len(args) != 1 {
+		return js.ValueOf(false)
+	}
+
+	if w.mac == nil {
+		return js.ValueOf(false)
+	}
+
+	// Convert JSON callback to grpc URI. JSON callbacks are of the form:
+	// `lnrpc.Lightning.WalletBalance` and the corresponding grpc URI is of
+	// the form: `/lnrpc.Lightning/WalletBalance`. So to convert the one to
+	// the other, we first convert all the `.` into `/`. Then we replace the
+	// first `/` back to a `.` and then we prepend the result with a `/`.
+	rg := regexp.MustCompile("(\\w+)\\.(\\w+)\\.(\\w+)")
+	uri := rg.ReplaceAllString(args[0].String(), "/$1.$2/$3")
+
+	ops, ok := perms[uri]
+	if !ok {
+		log.Errorf("uri %s not found in known permissions list", uri)
+		return js.ValueOf(false)
+	}
+
+	rawID := w.mac.Id()
+	if rawID[0] != byte(bakery.LatestVersion) {
+		log.Errorf("invalid macaroon version: %x", rawID)
+		return js.ValueOf(false)
+	}
+
+	decodedID := &lnrpc.MacaroonId{}
```

Comment:
We're using the `lnrpc` message here, so we don't need the whole `proto/macaroonpb` folder!

Reply:
omg ofc. Silly me 🙈  i see now that in the wasm-client it in any case already importing it from `lnrpc`. dont know where my mind was when i wrote this